src/core/json_object.h

```diff
@@ -8,6 +8,10 @@
 #include <optional>
 #include <string_view>
 
+extern "C" {
```

Comment:
why do you need it?

Reply:
Why do I need the `extern "C"` part?
Because it's a C-header, they have different linkage and namespace rules

Reply:
no, why do you need "redis/redis_aux.h" here?
OBJ_JSON is not used here and I believe that files that needs the types should include the types header

Reply:
I think of it as part of the exposed API of that file. The fact that it's defined elsewhere does not matter, users of `json_object.h` need it.
It's an implementation detail IMHO